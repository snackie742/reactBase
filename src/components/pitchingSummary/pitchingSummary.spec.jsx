import React from 'react';
import { shallow } from 'enzyme';
import PitchingSummary from './pitchingSummary';

const defaultProps = {
    winningPitcher: { player: { FirstName: 'Winning', LastName: 'Pitcher'}},
    losingPitcher: { player: { FirstName: 'Losing', LastName: 'Pitcher'}},
    savingPitcher: { player: { FirstName: 'Saving', LastName: 'Pitcher'}},

};

const setup = (propOverrides) => {
  const props = Object.assign({}, defaultProps, propOverrides);
  const wrapper = shallow(<PitchingSummary {...props} />);
  const component = wrapper.instance();
  return { wrapper, component };
};

describe('PitchingSummary', () => {
  describe('<PitchingSummary/>', () => {
    describe('(Snapshot)', () => {
      test('renders as expected with save', () => {
        const { wrapper } = setup(defaultProps);
        expect(wrapper).toMatchSnapshot();
      });
      test('renders as expected with without save', () => {
        const propOverrides = {
          winningPitcher: { player: { FirstName: 'Winning', LastName: 'Pitcher'}},
          losingPitcher: { player: { FirstName: 'Losing', LastName: 'Pitcher'}},
        };
        const { wrapper } = setup(propOverrides);
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
