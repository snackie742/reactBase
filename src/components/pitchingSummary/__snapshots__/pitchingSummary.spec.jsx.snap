// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PitchingSummary <PitchingSummary/> (Snapshot) renders as expected with save 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PitchingSummary
    losingPitcher={
        Object {
            "player": Object {
              "FirstName": "Losing",
              "LastName": "Pitcher",
            },
          }
    }
    savingPitcher={
        Object {
            "player": Object {
              "FirstName": "Saving",
              "LastName": "Pitcher",
            },
          }
    }
    winningPitcher={
        Object {
            "player": Object {
              "FirstName": "Winning",
              "LastName": "Pitcher",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="row"
>
          W - Winning Pitcher
</div>,
        <div
          className="row"
>
          L - Losing Pitcher
</div>,
        <div
          className="row"
>
          S - Saving Pitcher
</div>,
      ],
      "className": "col-lg pitchingSpacer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "W - Winning Pitcher",
          "className": "row",
        },
        "ref": null,
        "rendered": "W - Winning Pitcher",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "L - Losing Pitcher",
          "className": "row",
        },
        "ref": null,
        "rendered": "L - Losing Pitcher",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "S - Saving Pitcher",
          "className": "row",
        },
        "ref": null,
        "rendered": "S - Saving Pitcher",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="row"
>
            W - Winning Pitcher
</div>,
          <div
            className="row"
>
            L - Losing Pitcher
</div>,
          <div
            className="row"
>
            S - Saving Pitcher
</div>,
        ],
        "className": "col-lg pitchingSpacer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "W - Winning Pitcher",
            "className": "row",
          },
          "ref": null,
          "rendered": "W - Winning Pitcher",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "L - Losing Pitcher",
            "className": "row",
          },
          "ref": null,
          "rendered": "L - Losing Pitcher",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "S - Saving Pitcher",
            "className": "row",
          },
          "ref": null,
          "rendered": "S - Saving Pitcher",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`PitchingSummary <PitchingSummary/> (Snapshot) renders as expected with without save 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PitchingSummary
    losingPitcher={
        Object {
            "player": Object {
              "FirstName": "Losing",
              "LastName": "Pitcher",
            },
          }
    }
    savingPitcher={
        Object {
            "player": Object {
              "FirstName": "Saving",
              "LastName": "Pitcher",
            },
          }
    }
    winningPitcher={
        Object {
            "player": Object {
              "FirstName": "Winning",
              "LastName": "Pitcher",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="row"
>
          W - Winning Pitcher
</div>,
        <div
          className="row"
>
          L - Losing Pitcher
</div>,
        <div
          className="row"
>
          S - Saving Pitcher
</div>,
      ],
      "className": "col-lg pitchingSpacer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "W - Winning Pitcher",
          "className": "row",
        },
        "ref": null,
        "rendered": "W - Winning Pitcher",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "L - Losing Pitcher",
          "className": "row",
        },
        "ref": null,
        "rendered": "L - Losing Pitcher",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "S - Saving Pitcher",
          "className": "row",
        },
        "ref": null,
        "rendered": "S - Saving Pitcher",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="row"
>
            W - Winning Pitcher
</div>,
          <div
            className="row"
>
            L - Losing Pitcher
</div>,
          <div
            className="row"
>
            S - Saving Pitcher
</div>,
        ],
        "className": "col-lg pitchingSpacer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "W - Winning Pitcher",
            "className": "row",
          },
          "ref": null,
          "rendered": "W - Winning Pitcher",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "L - Losing Pitcher",
            "className": "row",
          },
          "ref": null,
          "rendered": "L - Losing Pitcher",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "S - Saving Pitcher",
            "className": "row",
          },
          "ref": null,
          "rendered": "S - Saving Pitcher",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
